proto/tendermint/abci/types.proto

```diff
@@ -186,16 +186,17 @@ message ResponseBeginBlock {
 }
 
 message ResponseCheckTx {
-  uint32         code       = 1;
-  bytes          data       = 2;
-  string         log        = 3;  // nondeterministic
-  string         info       = 4;  // nondeterministic
-  int64          gas_wanted = 5 [json_name = "gas_wanted"];
-  int64          gas_used   = 6 [json_name = "gas_used"];
-  repeated Event events     = 7 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
-  string         codespace  = 8;
-  string         sender     = 9;
-  int64          priority   = 10;
+  uint32         code          = 1;
+  bytes          data          = 2;
+  string         log           = 3;  // nondeterministic
+  string         info          = 4;  // nondeterministic
+  int64          gas_wanted    = 5 [json_name = "gas_wanted"];
+  int64          gas_used      = 6 [json_name = "gas_used"];
+  repeated Event events        = 7 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
+  string         codespace     = 8;
+  string         sender        = 9;
+  int64          priority      = 10;
+  string         mempool_error = 11;
```

Comment:
This change needs to be reflected in the spec. 

Also can you add a comment, does the application populate this field or should it be ignored?

This PR needs a changelog!